error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFlux.java

```diff
@@ -0,0 +1,106 @@
+
+/**
+ * A {@link BugChecker} that flags {@link reactor.core.publisher.Flux} operators that are implicitly
+ * blocking.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Avoid using `Flux` operators that implicitly block",
+    link = BUG_PATTERNS_BASE_URL + "ImplicitBlockingFlux",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = STYLE)
+public final class ImplicitBlockingFlux extends BugChecker implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<ExpressionTree> FLUX_WITH_IMPLICIT_BLOCK =
+      instanceMethod()
+          .onDescendantOf(Suppliers.typeFromString("reactor.core.publisher.Flux"))
+          .namedAnyOf("toIterable", "toStream")
+          .withNoParameters();
+  private static final Supplier<Type> STREAM = Suppliers.typeFromString("java.util.stream.Stream");
+
+  /** Instantiates a new {@link ImplicitBlockingFlux} instance. */
+  public ImplicitBlockingFlux() {}
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (!FLUX_WITH_IMPLICIT_BLOCK.matches(tree, state)) {
+      return Description.NO_MATCH;
+    }
+
+    Description.Builder description = buildDescription(tree);
+
+    description.addFix(SuggestedFixes.addSuppressWarnings(state, canonicalName()));
+    if (ThirdPartyLibrary.GUAVA.isIntroductionAllowed(state)) {
+      description.addFix(
+          trySuggestFix("com.google.common.collect.ImmutableList.toImmutableList", tree, state));
+    }
+    description.addFix(trySuggestFix("java.util.stream.Collectors.toList", tree, state));
+
+    return description.build();
+  }
+
+  private static SuggestedFix trySuggestFix(
+      String fullyQualifiedMethodInvocation, MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder importFix = SuggestedFix.builder();
+    String replacement =
+        SuggestedFixes.qualifyStaticImport(fullyQualifiedMethodInvocation, importFix, state);
+
+    return replaceWithExplicitBlock(tree, replacement + "()", importFix.build(), state);
+  }
+
+  private static SuggestedFix replaceWithExplicitBlock(
+      MethodInvocationTree tree,
+      String collectArgument,
+      SuggestedFix importFix,
+      VisitorState state) {
+    String collect = String.format("collect(%s)", collectArgument);
+    Types types = state.getTypes();
+    String explicitBlock =
+        types.isSubtype(ASTHelpers.getResultType(tree), types.erasure(STREAM.get(state)))
+            ? ".block().stream()"
+            : ".block()";
+    return replaceMethodInvocation(tree, collect, state)
+        .merge(importFix)
+        .postfixWith(tree, explicitBlock)
+        .build();
+  }
+
+  private static SuggestedFix.Builder replaceMethodInvocation(
+      MethodInvocationTree tree, String replacement, VisitorState state) {
+    String methodName = ASTHelpers.getSymbol(tree).getQualifiedName().toString();
```

Comment:
So for `MethodSymbol`s we can have a rule that rewrites `getQualifiedName` to `getSimpleName` because there will be no dots, right?

Reply:
That's one option, yes, though note that this also applies to other `Symbol` subtypes. A more generic rule (hence "or perhaps a simple bug checker") would be to perform this rewrite for any strict subtype of `Symbol` other than `{Class,Delegated,Package}Symbol`.

But in either case the question is whether this is worth the effort right now. It can/should go on the list of "things pointed out during code review" that we discussed last week, though :smile: